qa/L0_e2e/test_model.py

```diff
@@ -98,27 +109,46 @@ class GTILModel:
     """A compatibility wrapper for executing models with GTIL"""
 
     def __init__(self, model_path, model_format, output_class):
-        if model_format == "treelite_checkpoint":
+        if model_format == "xgboost":
+            self.tl_model = treelite.frontend.load_xgboost_model_legacy_binary(
+                model_path
+            )
+        elif model_format == "xgboost_json":
+            self.tl_model = treelite.frontend.load_xgboost_model(
+                model_path, format_choice="json"
+            )
+        elif model_format == "xgboost_ubj":
+            self.tl_model = treelite.frontend.load_xgboost_model(
+                model_path, format_choice="ubjson"
+            )
+        elif model_format == "lightgbm":
+            self.tl_model = treelite.frontend.load_lightgbm_model(model_path)
+        elif model_format == "treelite_checkpoint":
             self.tl_model = treelite.Model.deserialize(model_path)
-        else:
-            self.tl_model = treelite.Model.load(model_path, model_format)
         self.output_class = output_class
 
     def _predict(self, arr):
-        return treelite.gtil.predict(self.tl_model, arr)
+        result = treelite.gtil.predict(self.tl_model, arr)
+        # GTIL always returns prediction result with dimensions
+        # (num_row, num_target, num_class)
+        assert len(result.shape) == 3
```

Comment:
Why are we asserting anything in this method? This method is used just to get a "golden" value for comparison with results from Triton, right?

Reply:
In the future where we add multi-target models, the tests should fail so that we can revise them. In general, I like to use asserts to explicitly state assumptions being made about a test (here, we assume `n_outputs == 1`).